src/status_im/contexts/profile/push_notifications/effects.cljs

```diff
@@ -1,60 +1,130 @@
 (ns status-im.contexts.profile.push-notifications.effects
   (:require
     [native-module.push-notifications :as native-module.pn]
+    [promesa.core :as promesa]
     [react-native.platform :as platform]
-    [react-native.push-notification-ios :as pn-ios]
+    [status-im.common.log :as logger]
+    [status-im.config :as config]
+    [status-im.contexts.push-notifications.notifications :as pn-notifications]
+    [status-im.contexts.push-notifications.permissions :as pn-permissions]
+    [status-im.contexts.push-notifications.remote-token :as pn-remote-token]
     [utils.re-frame :as rf]))
 
-(def ios-listeners-added? (atom nil))
+(defn attempt-enable-notifications
+  []
+  (-> (pn-permissions/request-notification-permissions)
+      (promesa/then
+       (fn [permissions]
+         (if (:authorized? permissions)
+           permissions
+           (promesa/rejected
```

Comment:
I think you can just throw, but not sure if `promesa/rejected` is any different

Reply:
I think they're basically the same thing, I've tended to avoid throwing exceptions since throwing things has been problematic (for me) in the past. Though I imagine it's not a big throw since it's inside a promise chain.